File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string>words;
        string t="",reverse;
        for(auto&i:s){
            if(i!=' ')t+=i;
            else{
                if(t.size())words.push_back(t);
                t="";
            }
        }
        if(t.size())words.push_back(t);
        for(int i=words.size()-1;i>=1;i--)reverse+=words[i]+" ";
        reverse+=words[0];
        return reverse;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] collection = s.trim().split("\\s+");
        StringBuilder resultBuilder = new StringBuilder(); 
        for(int index =  collection.length - 1; index >= 0; --index){
            resultBuilder.append(collection[index]).append(" ");
        }
        return resultBuilder.toString().trim();
    }
}